### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by driehle**
- Marks 3.2 series as unmaintained
- Updates dependencies to latest versions
- Fixes one static analysis issue occuring with newer version of Psalm

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -53,34 +53,34 @@
...
         "container-interop/container-interop": "^1.2.0",
-        "doctrine/doctrine-module": "^5.0.0",
-        "doctrine/doctrine-laminas-hydrator": "^2.2.1 || ^3.0.0",
+        "doctrine/doctrine-module": "^5.1.0",
+        "doctrine/doctrine-laminas-hydrator": "^3.0.1",
         "doctrine/event-manager": "^1.1.1",
-        "doctrine/mongodb-odm": "^2.3.0",
-        "doctrine/persistence": "^2.2.3",
+        "doctrine/mongodb-odm": "^2.3.2",
+        "doctrine/persistence": "^2.4.0",
```
Comment (malarzm) [inline composer.json:60]: What's the point in bumping `doctrine/persistence`? [ODM itself is fine with ^2.2](https://github.com/doctrine/mongodb-odm/blob/2.3.x/composer.json#L32)

File: composer.json [hunk 2]
```diff
@@ -53,34 +53,34 @@
         "php": "^7.4 || ~8.0.0 || ~8.1.0",
         "ext-mongodb": "*",
         "container-interop/container-interop": "^1.2.0",
-        "doctrine/doctrine-module": "^5.0.0",
-        "doctrine/doctrine-laminas-hydrator": "^2.2.1 || ^3.0.0",
+        "doctrine/doctrine-module": "^5.1.0",
+        "doctrine/doctrine-laminas-hydrator": "^3.0.1",
```
Comment (malarzm) [inline composer.json:57]: What's the point in dropping ^2.2.1? That's really drastic for a bugfix release

**Review by malarzm — CHANGES_REQUESTED** [review]
Left some questions but I believe these changes are way too drastic for a bugfix release

File: composer.json (hunk 2 above)
Comment (driehle) [reply composer.json:57]: 2.2 series will soon be unmaintained

File: composer.json (hunk 1 above)
Comment (driehle) [reply composer.json:60]: Why not? Since it is the same major there shouldn't be any issues in upgrading.

Comment (Ocramius) [reply composer.json:60]: Probably best done in a minor release, rather than a patch release

Comment (driehle) [reply composer.json:60]: Ok, I changed the base

**Review by malarzm — APPROVED** [review]
I'm perfectly fine with these bumps in a minor release 👍

**Pull request merged**